src/native/libs/System.Native/pal_io.c

```diff
@@ -1952,17 +1952,34 @@ int32_t SystemNative_PWrite(intptr_t fd, void* buffer, int32_t bufferSize, int64
 #if (HAVE_PREADV || HAVE_PWRITEV) && !defined(TARGET_WASM)
 static int GetAllowedVectorCount(int32_t vectorCount)
```

Comment:
Given `IOV_MAX` is POSIX, we can delete this method and inline its usage. The only non-POSIX system we support is Windows.

To fix mac issue we can use batching; in SystemNative_PWriteV()..

```c
#if HAVE_PWRITEV && !defined(TARGET_WASM)
    size_t totalLength = 0;
    for (int i = 0; i < vectorCount; i++) 
    {
        totalLength += vectors[i].Count;
    }

    // If the total length exceeds INT_MAX, batch the vectors into smaller groups
    if (totalLength > INT_MAX) 
    {
        while (vectorCount > 0) 
        {
            size_t batchLength = 0;
            int batchCount = 0;

            // Determine the maximum number of vectors that fit within the INT_MAX limit
            while (batchCount < vectorCount && batchLength + vectors[batchCount].Count <= INT_MAX) 
            {
                batchLength += vectors[batchCount].Count;
                batchCount++;
            }

            // Perform the write operation for the current batch
            while ((count = pwritev(fileDescriptor, (struct iovec*)vectors, batchCount, (off_t)fileOffset)) < 0 && errno == EINTR);

            if (count < 0)
                return count;

            // Update vector count, pointers, and file offset for the next batch
            vectorCount -= batchCount;
            vectors += batchCount;
            fileOffset += count;
        }
    } 
    else 
    {
        // Perform the write operation for the entire set of vectors if the total length is within the limit
        while ((count = pwritev(fileDescriptor, (struct iovec*)vectors, IOV_MAX, (off_t)fileOffset)) < 0 && errno == EINTR);
    }
#else
```